client-server-server/src/main/scala/zio/raft/server/RaftServer.scala

```diff
@@ -0,0 +1,471 @@
+                  _ <- transport.sendMessage(routingId, SessionContinued(nonce)).orDie
+                  newSessions = sessions.reconnect(sessionId, routingId, now)
+                } yield copy(sessions = newSessions)
+              case None =>
+                transport.sendMessage(routingId, SessionRejected(RejectionReason.SessionNotFound, nonce, None)).orDie.as(this)
+            }
+          
+          case KeepAlive(timestamp) =>
+            sessions.findSessionByRouting(routingId) match {
+              case Some(sessionId) =>
+                for {
+                  now <- Clock.instant
+                  _ <- transport.sendMessage(routingId, KeepAliveResponse(timestamp)).orDie
+                  newSessions = sessions.updateExpiry(sessionId, now)
+                } yield copy(sessions = newSessions)
+              case None =>
+                transport.sendMessage(routingId, SessionClosed(SessionCloseReason.SessionError, None)).orDie.as(this)
+            }
+          
+          case clientRequest: ClientRequest =>
+            sessions.findSessionByRouting(routingId) match {
+              case Some(sessionId) =>
+                for {
+                  _ <- raftActionsOut.offer(RaftAction.ClientRequest(sessionId, clientRequest.requestId, clientRequest.payload))
+                } yield this
+              case None =>
+                transport.sendMessage(routingId, RequestError(RequestErrorReason.SessionTerminated, None)).orDie.as(this)
+            }
+          
+          case ServerRequestAck(requestId) =>
+            // Could track acknowledgments if needed
+            ZIO.succeed(this)
+          
+          case CloseSession(reason) =>
+            sessions.findSessionByRouting(routingId) match {
+              case Some(sessionId) =>
+                for {
+                  _ <- ZIO.logInfo(s"Client closed session $sessionId: $reason")
+                  newSessions = sessions.disconnect(sessionId, routingId)
+                } yield copy(sessions = newSessions)
+              case None =>
+                ZIO.succeed(this)
+            }
+        }
+      }
+    }
+  }
+  
+  /**
+   * Immutable sessions management.
+   */
+  case class Sessions(
+    metadata: Map[SessionId, SessionMetadata],
+    connections: Map[SessionId, SessionConnection],
+    routingToSession: Map[RoutingId, SessionId]
+  ) {
+    def addConnected(sessionId: SessionId, routingId: RoutingId, capabilities: Map[String, String], now: Instant): Sessions = {
+      val expiresAt = now.plusSeconds(90) // TODO: use config
+      copy(
+        metadata = metadata.updated(sessionId, SessionMetadata(capabilities, now)),
+        connections = connections.updated(sessionId, SessionConnection(Some(routingId), expiresAt)),
+        routingToSession = routingToSession.updated(routingId, sessionId)
+      )
+    }
+    
+    def reconnect(sessionId: SessionId, routingId: RoutingId, now: Instant): Sessions = {
+      connections.get(sessionId) match {
```

Comment:
we can accidentally leave an old mapping from routing id to same session id.
So if the connection has routing id, remove it from routingToSession.